src/numerics/include/metaphysicl/dualnumber_decl.h

```diff
@@ -222,14 +239,32 @@ operator opname (DualNumber<T,D>&& a, const T2& b); \
         DualNumber_decl_preop(opname, functorname)
 #endif
 
-
 DualNumber_decl_op(+, Plus)
 DualNumber_decl_op(-, Minus)
 DualNumber_decl_op(*, Multiplies)
 DualNumber_decl_op(/, Divides)
 
-
-#define DualNumber_decl_compare(opname) \
+#define DualNumber_NumberArray_decl_op(opname, functorname)                                        \
```

Comment:
Would it make sense to move these to dualnumberarray.h?  Handling interactions between the two was the specific purpose of that header.  I'm not happy with my solution for the obvious combinatorial-explosion reasons but it at least maintains a *little* modularity.